bpfd/src/lib.rs

```diff
@@ -144,43 +145,50 @@ pub async fn serve(config: Config, static_programs: Vec<StaticPrograms>) -> anyh
                         proceed_on
                     };
 
-                    let prog_data =
+                    let prog_data_result =
                         ProgramData::new_from_location(location, section_name.clone(), username)
-                            .await?;
-                    let prog = match program_type {
-                        command::ProgramType::Xdp => Program::Xdp(XdpProgram {
-                            data: prog_data.clone(),
-                            info: NetworkMultiAttachInfo {
-                                if_index,
-                                current_position: None,
-                                metadata: command::Metadata {
-                                    priority,
-                                    // This could have been overridden by image tags
-                                    name: prog_data.section_name,
-                                    attached: false,
-                                },
-                                proceed_on: proc_on,
-                                if_name: iface,
-                            },
-                        }),
-                        command::ProgramType::Tc => Program::Tc(TcProgram {
-                            data: prog_data.clone(),
-                            info: NetworkMultiAttachInfo {
-                                if_index,
-                                current_position: None,
-                                metadata: command::Metadata {
-                                    priority,
-                                    name: prog_data.section_name,
-                                    attached: false,
-                                },
-                                proceed_on: proc_on,
-                                if_name: iface,
-                            },
-                            direction: direction.unwrap(),
-                        }),
-                        _ => panic!("unsupported prog type"),
-                    };
-                    bpf_manager.add_program(prog)
+                            .await;
+
+                    match prog_data_result {
+                        // TODO: Return better error for Err case?
+                        Err(e) => Err(BpfdError::Error(format!("{e}"))),
```

Comment:
I think this will work alright maybe just add an error like 

```
    #[error("Failed to get BpfBytecode {0}")]
    BpfBytecodeError(#[from] ParseError)
 ```
 
 then `Err(e) => Err(BpfdError::BpfBytecodeError(e))` Should work (If we can pass the compiler :)